app/scripts/controllers/transactions/lib/tx-state-history-helpers.js

```diff
@@ -35,11 +35,16 @@ export function generateHistoryEntry(previousState, newState, note) {
   const entry = jsonDiffer.compare(previousState, newState);
   // Add a note to the first op, since it breaks if we append it to the entry
   if (entry[0]) {
+    entry[0].timestamp = Date.now();
     if (note) {
       entry[0].note = note;
+      if (
+        note.toLowerCase().includes('setting status to confirmed') ||
+        note.toLowerCase().includes('transactions#confirmtransaction')
+      ) {
+        entry[0].timestamp = newState.time;
```

Comment:
I think this will work if the transaction is mined within the same minute that it is created within MetaMask, but I don't think it will work if the mining happens significantly later. This is hard to test on test networks, where the mining time is short.

Reply:
I see mmmh